packages/grafana-ui/src/components/DataSourceSettings/Sigv4AuthSettings.tsx

```diff
@@ -0,0 +1,242 @@
+import React from 'react';
+import { HttpSettingsProps } from './types';
+import { SelectableValue } from '@grafana/data';
+import { Button, InlineFormLabel, Input } from '..';
+import Select from '../Forms/Legacy/Select/Select';
+
+export const Sigv4AuthSettings: React.FC<HttpSettingsProps> = props => {
+  const { dataSourceConfig } = props;
+
+  const authProviderOptions = [
+    { label: 'Access & secret key', value: 'keys' },
+    { label: 'Credentials file', value: 'credentials' },
+    { label: 'ARN', value: 'arn' },
```

Comment:
There is a [PR](url) for refactoring the respective Cloudwatch authentication where ARN [will not be a separate authentication provider](https://github.com/grafana/grafana/pull/25548/files#diff-d554d037cd36865918be4c50e78ec9ddR19).
Maybe we want to have the same approach here as well
@aknuds1 what do you think?

Reply:
@papagian I think you are right, assuming that the approach in the [PR](https://github.com/grafana/grafana/pull/25548) is correct. I'm not 100% on top of that one right now, but I think it's correct that assuming a role isn't an authentication type of itself, but something additional.

Reply:
Yeah I made a comment about this previously. Happy to wait merging this until that PR is figured out 😄